testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppPermissionAdapter.kt

```diff
@@ -28,53 +26,46 @@ class PreloadMiniAppPermissionAdapter :
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.permissionName.text = toReadableName(manifestPermissionNames[position])
+        holder.permissionName.text = toReadableName(manifestPermissions[position].type)
+        holder.permissionSwitch.visibility =
+            if (manifestPermissions[position].isRequired) View.GONE else View.VISIBLE
+        holder.permissionStatus.visibility =
+            if (manifestPermissions[position].isRequired) View.VISIBLE else View.GONE
+
         holder.permissionSwitch.isChecked =
```

Comment:
The "optional" permissions should default to `true` if the permission hasn't been requested before - i.e. the first time that the mini app is downloaded.